server/me.go

```diff
@@ -111,28 +176,49 @@ func (s *Service) Me(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if usr != nil {
+		usr.CurrentOrganization = p.Organization
 		res := newMeResponse(usr)
 		encodeJSON(w, http.StatusOK, res, s.Logger)
 		return
 	}
 
 	// Because we didnt find a user, making a new one
 	user := &chronograf.User{
-		Name:     username,
-		Provider: provider,
+		Name:     p.Subject,
+		Provider: p.Issuer,
 		// TODO: This Scheme value is hard-coded temporarily since we only currently
 		// support OAuth2. This hard-coding should be removed whenever we add
 		// support for other authentication schemes.
-		Scheme: "oauth2",
+		Scheme: scheme,
+		Roles: []chronograf.Role{
+			{
+				Name: MemberRoleName,
+				// This is the ID of the default organization
+				Organization: "0",
+			},
+		},
+		SuperAdmin: s.firstUser(),
 	}
 
-	newUser, err := s.UsersStore.Add(ctx, user)
+	newUser, err := s.Store.Users(ctx).Add(ctx, user)
 	if err != nil {
 		msg := fmt.Errorf("error storing user %s: %v", user.Name, err)
 		unknownErrorWithMessage(w, msg, s.Logger)
 		return
 	}
 
+	newUser.CurrentOrganization = p.Organization
 	res := newMeResponse(newUser)
 	encodeJSON(w, http.StatusOK, res, s.Logger)
 }
+
+// TODO(desa): very slow
+func (s *Service) firstUser() bool {
```

Comment:
Add FinalDelegate or somesuch  to Rancor

Reply:
Opened https://github.com/influxdata/chronograf/issues/2211